shell/platform/android/android_context_gl.cc

```diff
@@ -204,14 +204,18 @@ AndroidContextGL::~AndroidContextGL() {
 
 std::unique_ptr<AndroidEGLSurface> AndroidContextGL::CreateOnscreenSurface(
     fml::RefPtr<AndroidNativeWindow> window) const {
-  EGLDisplay display = environment_->Display();
+  if (window->IsFakeWindow()) {
+    return CreatePbufferSurface();
+  } else {
```

Comment:
Why do we need to create a PBuffer surface? 

How confident can we be that we're testing what we even care about here?

Is there any possible way we could avoid this code in production code?

Reply:
The tests in this PR are just very simple setup and teardown.  This is more about setting up the infrastructure to make more interesting tests that do assert more things.

Reply:
As far as the IsFakeWindow question, I looked into it.  I didn't see a way to do it that wasn't a significant refactor.  That's sometimes the problem when you add tests after the fact versus build them up with tests.

You could transfer creating the surface to the window, but that would leak egl calls outside of the AndroidContextGL.  I'm open to any suggestion I might have overlooked.

Reply:
> Why do we need to create a PBuffer surface?

The code assumes a surface, it will crash if you don't have one.  We don't have a window so our only option is to make a fake pbuffer surface.